homeassistant/components/datadog/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for Datadog."""
+
+from typing import Any
+
+from datadog import DogStatsd
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult, OptionsFlow
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_PREFIX
+from homeassistant.core import HomeAssistant, callback
+
+from .const import (
+    CONF_RATE,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_PREFIX,
+    DEFAULT_RATE,
+    DOMAIN,
+)
+
+
+class DatadogConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Datadog."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle user config flow."""
+        errors: dict[str, str] = {}
+        if user_input:
+            # Validate connection to Datadog Agent
+            success = await validate_datadog_connection(
+                self.hass,
+                user_input,
+            )
+            if not success:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title=f"Datadog {user_input['host']}", data={}, options=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+                    vol.Required(CONF_PREFIX, default=DEFAULT_PREFIX): str,
+                    vol.Required(CONF_RATE, default=DEFAULT_RATE): int,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config: dict) -> ConfigFlowResult:
+        """Handle import from configuration.yaml."""
+        return await self.async_step_user(import_config)
```

Comment:
The import flow should not return errors. Instead it should abort. We should also create a repair issue explaining the user that they should remove the YAML config, or, if it couldn't connect, tell them they couldnt import because we couldnt commit.

I don't think there are many integrations which have it implemented right now, so be sure to check for merged PRs with the config-flow label